### PR TITLE
Exit early in case of failure, refactor testing results into a function

### PR DESCRIPTION
**Pull request by christiam**

### CODE REVIEW

File: submit-and-wait-for-results.sh [hunk 1]
```diff
@@ -51,6 +51,32 @@ cleanup_resources_on_error() {
     exit 1;
 }
 
+check_results() {
+    db=`awk '/^db/ {print $NF}' $CFG`
+    if compgen -G "batch*$db.out.gz" >/dev/null ; then
+        test $(du -a *.out.gz | sort -n | head -n 1 | cut -f 1) -gt 0
```
Comment (boratyng) [inline submit-and-wait-for-results.sh:57]: Tabular output (-outfmt 6) will be empty if there are no results.

File: submit-and-wait-for-results.sh [hunk 2]
```diff
@@ -51,6 +51,32 @@ cleanup_resources_on_error() {
...
+                echo "FATAL ERROR: Found mismatched results: configured $db, actual $searched_db"
+                exit 1
+            fi
+            find . -maxdepth 1 -name "batch*$db.out.gz" -type f -print0 | xargs -0 -P $NTHREADS zcat | \
+                awk 'BEGIN{t=0} /hits found/ {t+=$2} END{print "Total hits found", t}'
+            num_hsps=`find . -maxdepth 1 -name "batch*$db.out.gz" -type f -print0 | xargs -0 zcat | grep -v '^#' | wc -l`
+            echo "Number of HSPs found $num_hsps"
+        elif grep -q 'outfmt 6' $logfile; then
+            num_hsps=`find . -maxdepth 1 -name "batch*$db.out.gz" -type f -print0 | xargs -0 zcat | grep -v '^#' | wc -l`
+            echo "Number of HSPs found $num_hsps"
```
Comment (boratyng) [inline submit-and-wait-for-results.sh:73]: Do users know what an HSP is? Should it be results or hits?

File: submit-and-wait-for-results.sh (hunk 1 above)
Comment (christiam) [reply submit-and-wait-for-results.sh:57]: Good point, thanks, I'll look into this.

File: submit-and-wait-for-results.sh (hunk 2 above)
Comment (christiam) [reply submit-and-wait-for-results.sh]: I don't know, but we can spell out what an HSP is: 3920839

Comment (boratyng) [reply submit-and-wait-for-results.sh]: I see the same problem with spelled out HSPs. Why not call them search results or database hits. I think these are terms that most people will understand.

Also, this is the definition of an HSP from https://www.ncbi.nlm.nih.gov/books/NBK62051/, linked from https://blast.ncbi.nlm.nih.gov/Blast.cgi?CMD=Web&PAGE_TYPE=BlastDocs:
```
HSP
A High-scoring Segment Pair (HSP) is a local alignment with no gaps that achieves one of the highest alignment scores in a given search.
```
Then an HSP is not the same as a final search result that may have gaps.

**Review by boratyng — APPROVED** [review]

**Pull request merged**